### DISCUSSION TITLE
Behaviour of selector operators >> and <<

### DISCUSSION THREAD

**Discussion by voneiden** [gumyr/build123d#447] [General]
Build123d has handy selector operators as detailed in https://build123d.readthedocs.io/en/latest/key_concepts.html#quick-reference

The operators `>>` and `<<` behave different from the rest, as they implicitly return the first result. To me it feels like a design oversight. I've tripped on it twice myself. First time because I assumed `>>` and `<<` work like `>` and `<` operators and second time because I had forgotten that they don't. Also, at least personally, I do often find myself needing to select a group that is not the first one and that degrades the utility value of this operator. 

Thoughts? 

Changing the behaviour at this point of course shouldn't be taken lightly as it would be a breaking change.

**Comment by gumyr** [comment]
Your thoughts here are very valid; however, `>>` has come to mean "top" or "bottom" as follows:
```py
from build123d import *
from ocp_vscode import *

with BuildPart() as ex:
    Box(10, 10, 10)
    with BuildSketch(ex.faces() >> Axis.Z):
        Circle(3)
    extrude(amount=3)

show(ex)
```
![image](https://github.com/gumyr/build123d/assets/85638518/e3a48a37-8df0-4bed-86c5-6503feab4197)
as this is a very common use case.  

Personally I don't use the select operators much because the order of operations can't be changed which results in having to put parentheses all over the place.
